internal/storage/storage.go

```diff
@@ -1,312 +1,14 @@
 package storage
 
 import (
-	"bytes"
-	"compress/gzip"
 	"context"
-	"crypto/sha256"
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"sort"
+	"io/fs"
 
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
-	"sigs.k8s.io/yaml"
-
-	"github.com/operator-framework/rukpak/internal/util"
 )
 
 type Storage interface {
-	Load(ctx context.Context, owner client.Object) ([]unstructured.Unstructured, error)
-	Store(ctx context.Context, owner client.Object, objects []client.Object) error
-}
-
-var _ Storage = &ConfigMaps{}
-
-type ConfigMaps struct {
-	Client     client.Client
-	Namespace  string
-	NamePrefix string
+	Load(ctx context.Context, owner client.Object) (fs.FS, error)
+	Store(ctx context.Context, owner client.Object, bundle fs.FS) error
+	Delete(ctx context.Context, owner client.Object) error
```

Comment:
This looks good.  Can we support multiple kind of storage and make them selectable?

Reply:
How would a provisioner expose configuring storage? I assume it would be through the ProvisionerClass API, but we'd need to be able to query for that resource in a main.go before the manager instance has started.